jl4-lsp/src/LSP/L4/Actions.hs

```diff
@@ -139,7 +140,7 @@ visualise mtcRes (getRecVis, setRecVis) uri msrcPos = do
   case mdecide of
     Nothing -> pure (InR Null)
     Just (decide, simp, substitution) ->
-      case Ladder.doVisualize decide (Ladder.MkVizEnv (toNormalizedUri uri) substitution simp) of
+      case Ladder.doVisualize decide (Ladder.mkVizEnv verTextDocId substitution simp) of
```

Comment:
I think I would have personally (so no need to change it) not have introduced this smart constructor, because it doesn't "do" anything, it would've been easier to read to just have `Ladder.MkVizEnv verTextDocId._uri ...`